scripts/run_mcmc.py

```diff
@@ -0,0 +1,295 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Runs the MCMC sampler for tree inference
+Using the given data, error rates and initial tree.
+
+As per definition in the SCITE Jahn et al. 2016.
+
+Allows to start an MCMC chain from a given tree provided as
+a dumped tree json file or generate a random tree.
+
+Example Usage:
+poetry run python ../scripts/run_mcmc.py
+
+Example Usage with arguments:
+    poetry run python scripts/run_mcmc.py
+    --config_fp data/config/mcmc_config_mark00.json
+    --out_dir data/mcmc/mark01
+    --data_fp data/mock/seed_32_n_..._tree_3.json
+
+    if you want to provide a tree:
+    from a mcmc run:
+    --init_tree_fp data/mcmc/mark01/samples_XXXXXXXX_XXXXXX.json
+    --iteration 1000
+    or to read in a TreeNode
+    --init_tree_fp data/trees/tree_3.json
+    --init_TreeNode
+
+Config File .json:
+{
+    "move_probs": {
+        "prune_and_reattach": 0.1,
+        "swap_node_labels": 0.65,
+        "swap_subtrees": 0.25
+    },
+    "fnr": 0.1,
+    "fpr": 0.2,
+    "num_samples": 1000,
+    "burn_in": 20,
+    "thinning": 100
+}
+
+"""
+
+import argparse
+import jax.random as random
+import json
+import jax.numpy as jnp
+import logging
+
+from pathlib import Path
+from datetime import datetime
+
+import pyggdrasil.tree_inference as tree_inf
+import pyggdrasil.serialize as serialize
+
+from pyggdrasil.tree_inference import (
+    MutationMatrix,
+    mcmc_sampler,
+    MoveProbabilities,  # type: ignore
+)
+
+
+def create_parser() -> argparse.Namespace:
```

Comment:
Very nice, I see a lot of progress in writing parsers on your side!

Reply:
Thank you - this means a lot to me!